### PR TITLE
vkectl 0.1.1 (new formula)

### PR DESCRIPTION
**Pull request by xpofei**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/vkectl.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/volcengine/vkectl/pkg/version.version=v#{version}"), "./main"
+  end
+
+  test do
+    version_out = shell_output("#{bin}/vkectl version")
+    assert_match version.to_s, version_out
+
+    resource_help_out = shell_output("#{bin}/vkectl resource -h")
```
Comment (carlocab) [inline Formula/vkectl.rb:18]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for the PR.

File: Formula/vkectl.rb (hunk 1 above)
Comment (xpofei) [reply Formula/vkectl.rb:18]: PTAL

File: Formula/vkectl.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    assert_match "AddNodes", resource_help_out
+    assert_match "AddVciSubnets", resource_help_out
+    assert_match "ListClusters", resource_help_out
+    assert_match "CheckCidrConflict", resource_help_out
+    assert_match "CreateNodePool", resource_help_out
```
Comment (carlocab) [inline Formula/vkectl.rb:23]:
```suggestion
```
We don't need to check this much, as then we run the risk of making this too fragile with respect to upstream changes.

File: Formula/vkectl.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    resource_get_addon_out = shell_output("#{bin}/vkectl resource GetAddon")
+    assert_match '{"Name":"","Version":"","Status":"","ErrorMessage":""}', resource_get_addon_out
+
+    security_get_check_item_out = shell_output("#{bin}/vkectl security GetCheckItem")
+    assert_match '{"Number":"","Description":"","Rationale":"","Remediation":"","Impact":""}',
+      security_get_check_item_out
```
Comment (carlocab) [inline Formula/vkectl.rb:30]: Is this supposed to be JSON? If so, we can just do something like
```suggestion
    resource_get_addon_out = JSON.parse(shell_output("#{bin}/vkectl resource GetAddon"))
    assert_empty resource_get_addon_out["Name"]

    security_get_check_item_out = JSON.parse(shell_output("#{bin}/vkectl security GetCheckItem"))
    assert_empty security_get_check_item_out["Number"]
```
That way we just check it's valid JSON and check one field each for what we expect here.

**Review by carlocab — COMMENTED** [review]
Thanks; this is looking much better. Just a few more suggestions.

File: Formula/vkectl.rb (hunk 3 above)
Comment (xpofei) [reply Formula/vkectl.rb:26]: yeah, it will indeed only be JSON, thanks for the suggestion

**Comment by xpofei** [comment]
> Thanks; this is looking much better. Just a few more suggestions.

hi, @carlocab . PTAL

File: Formula/vkectl.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  test do
+    version_out = shell_output("#{bin}/vkectl version")
+    assert_match version.to_s, version_out
+
+    resource_help_out = shell_output("#{bin}/vkectl resource -h")
+    assert_match "AddNodes", resource_help_out
+
+    resource_get_addon_out = shell_output("#{bin}/vkectl resource GetAddon")
+    resource_get_addon_index = resource_get_addon_out.index("{")
```
Comment (carlocab) [inline Formula/vkectl.rb:22]: It seems weird to need this. Is it because the initial part of the output is not valid JSON?

Comment (xpofei) [reply Formula/vkectl.rb:22]: yeah, the output contains two parts, the second part is the json format

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**